src/shogun/distance/MahalanobisDistance.cpp

```diff
@@ -39,20 +37,43 @@ bool CMahalanobisDistance::init(CFeatures* l, CFeatures* r)
 {
 	CRealDistance::init(l, r);
```

Comment:
instead of the REQUIRED sections it should be just check the return type of `CRealDistance::init(l, r)` as it checks for everything you need want....